extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/ExtendedCharactersSupport.java

```diff
@@ -5,13 +5,13 @@
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.pkg.NativeConfig;
-import io.quarkus.deployment.pkg.steps.NativeBuild;
+import io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild;
 import io.quarkus.jdbc.oracle.runtime.OracleInitRecorder;
 
 public final class ExtendedCharactersSupport {
 
     @Record(STATIC_INIT)
-    @BuildStep(onlyIf = NativeBuild.class)
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
```

Comment:
I think this is going to be confusing. We should at least add a comment in `NativeBuild` to tell it should be used in very specific cases and the other should be preferred. But I'm even wondering if we should merge the two and always in `NativeBuild`. Or do we envision having to run something only for the native compilation and not the native sources build?

Reply:
Agreed about the comment, but we can't merge the two as there are cases where we need to differentiate (like in the container-image extensions)

Reply:
I think I would rather have `NativeBuild` targeting the two and having something like `NativeBuildOnly` or something similar for the other.
Otherwise my guess is that we will keep hitting that wall.

Reply:
I'm -0.5 on that. I won't make that change myself, but won't object to it being made by anyone else

Reply:
OK, well, I suppose we will see how it goes then. Please add a big warning in the Javadoc though.

Reply:
Will do

Reply:
Done